sample_apps/run_example.sh

```diff
@@ -641,10 +641,16 @@ function run_popd() {
 # Deep-clean the build-env to remove artifacts (e.g. generated files, binaries
 # etc.) that may have been produced by other steps. We run this to ensure
 # that this script will run successfully w/o any dependencies on executing
-# some prior steps.
+# some prior steps. This function will rm files that are listed in the
+# .gitignore file for this project. This way we won't delete any user's
+# config-/env-files saved for this project.
 # ###########################################################################
 function rm_non_git_files() {
     run_cmd
+    echo "${Me}: Delete all files not tracked by git that are also ignored."
+    if [ $DryRun -eq 1 ]; then
+        return
```

Comment:
Fixes an existing but that even in `--dry-run` mode this command was actually doing some work and deleting files. 

It should not do anything in dry-run mode.